lib/activerecord-multi-tenant/relation_extension.rb

```diff
@@ -7,29 +7,55 @@ def delete_all
       # Call the original delete_all method if the current tenant is identified by an ID
       return super if MultiTenant.current_tenant_is_id? || MultiTenant.current_tenant.nil?
 
+      stmt = Arel::DeleteManager.new.from(table)
+      stmt.wheres = [generate_in_condition_subquery]
+
+      # Execute the delete statement using the connection and return the result
+      klass.connection.delete(stmt, "#{klass} Delete All").tap { reset }
+    end
+
+    # Overrides the update_all method to include tenant scoping
+    def update_all(updates)
+      # Call the original update_all method if the current tenant is identified by an ID
+      return super if MultiTenant.current_tenant_is_id? || MultiTenant.current_tenant.nil?
+
+      stmt = Arel::UpdateManager.new
+      stmt.table(table)
+      stmt.set Arel.sql(@klass.send(:sanitize_sql_for_assignment, updates))
```

Comment:
@Amit909Singh @gurkanindibay  In original Rails 7.2 implementation here is  condition to `if updates.is_a?(Hash)` and then there is different logic:

https://github.com/rails/rails/blob/69c86f9606e490b24e8628dc47f34fbec6b321d1/activerecord/lib/active_record/relation.rb#L581

This now creates issue for me when using Rails 7.2 with `counter_cache` and I tracked it down to this line.
In Rails 7.2 the condition looks like `column_count = COALESCE(column_count, 0) + 1` because it is not sanitized with `sanitize_sql_for_assignment`.

Here with multitenant gem it is always sanitized with `sanitize_sql_for_assignment` which creates `column_count = NULL` query.

Do you think we should copy-paste the logic from the original Rails's update_all method?